STM32Cube/Tasks/can_handler.c

```diff
@@ -0,0 +1,38 @@
+#include "can_handler.h"
+
+
+uint32_t LED_state = 0;
+xQueueHandle busQueue;
+
+void can_handle_rx(const can_msg_t *message, uint32_t timestamp){
+	if(get_message_type(message) == MSG_LEDS_ON)
+	{
+		HAL_GPIO_WritePin(GPIOB, GPIO_PIN_5, 0);
+	}
+	else if (get_message_type(message) == MSG_LEDS_OFF)
+	{
+		HAL_GPIO_WritePin(GPIOB, GPIO_PIN_5, 1);
+	}
+	return;
+}
+
+void canHandlerTask(void *argument)
+{
+  for(;;)
+  {
+	can_msg_t tx_msg;
+	//Block the thread until we see data in the bus queue or 5 ticks elapse
+	if(xQueueReceive(busQueue, &tx_msg, 10) == pdTRUE) //Returns pdTRUE if we got a message, pdFALSE if timed out
```

Comment:
would it better to make this timeout (practically) indefinite instead of 10? Since there's no action to take if queue never receives it should just be blocked (practically) indefinitely.
Unless you want some error handling if msg isnt received every 10 ticks.